frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplateDetails/ClusterTemplateDetailsPage.spec.tsx

```diff
@@ -0,0 +1,89 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import { render } from '@testing-library/react';
+import { Route, Router, Switch } from 'react-router-dom';
+import { createMemoryHistory } from 'history';
+import ClusterTemplateDetailsPage from './ClusterTemplateDetailsPage';
+import { useClusterTemplate } from '../hooks/useClusterTemplates';
+import exampleTemplate from '../mocks/clusterTemplateExample.json';
+
+jest.mock('../hooks/useClusterTemplates');
+jest.mock('../hooks/useQuotas', () => {
+  return {
+    useQuotas: jest.fn().mockReturnValue([[], true, null]),
+  };
+});
+jest.mock('../hooks/useClusterTemplateInstances', () => {
+  return {
+    useClusterTemplateInstances: () => [[], true, null],
+  };
+});
+
+jest.mock('@openshift-console/dynamic-plugin-sdk', () => {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  return {
+    ResourceLink: jest.fn().mockImplementation(({ name }) => <div>{name}</div>),
```

Comment:
Alternatively you can use `MockComponent`

```suggestion
  const MockComponent = require('../mocks/MockComponent').default;
  return {
    ResourceLink: MockComponent,
```

Reply:
The idea was that mocking ResourceLink this way I can verify that the name was sent correctly, by later verifying the text content of the mocked component

Reply:
In the end I figured out how to use it